source/ota_mqtt.c

```diff
@@ -894,9 +899,13 @@ OtaErr_t requestJob_Mqtt( const OtaAgentContext_t * pAgentCtx )
     ( void ) pOtaJobsGetNextTopicTemplate;
     ( void ) pOtaGetNextJobMsgTemplate;
 
+    /* Client token max length is 64. It is a combination of request counter (max 10 characters), a separator colon, and the ThingName. */
+    xThingNameLength = strlen( ( const char * ) pAgentCtx->pThingName );
+    ( void ) strncpy( pcClientTokenThingName, ( const char * ) pAgentCtx->pThingName, ( xThingNameLength > OTA_CLIENT_TOKEN_MAX_THINGNAME_LEN ) ? OTA_CLIENT_TOKEN_MAX_THINGNAME_LEN : xThingNameLength );
+
     pTopicParts[ 1 ] = ( const char * ) pAgentCtx->pThingName;
     pPayloadParts[ 1 ] = reqCounterString;
-    pPayloadParts[ 3 ] = ( const char * ) pAgentCtx->pThingName;
+    pPayloadParts[ 3 ] = ( const char * ) pcClientTokenThingName;
```

Comment:
Instead of declaring the new `pcClientTokenThingName` and `xThingNameLength` variables to use to assign here would it be possible to use `strncpy()` call here and directly copy the values into `pPayloadParts[3]`? Just to try and reduce  the memory footprint?

Reply:
I like this suggestion. The only issue I see with applying it is that null-terminating the copied string will become more difficult.